[typescript-react-apollo] Add support for importing document node from external file for hooks and HOCs

eddeee888 opened the pull request:
This PR adds support for importing document node from an external file for hooks i.e. `useQuery`, `useLazyQuery`, `useMutation`, `useSubscription`. 

The current implementation only works with `withComponent` config. As the result, hook document nodes are not used correctly. This can be seen here: https://codesandbox.io/s/gcg-15-experiments-c8l15 (Look for `src/GetProject.operation.generated.tsx`)

dotansimha commented:
@eddeee888 thank you!!!
we can apply it to all available options? (hooks/components/hocs) ?

eddeee888 commented:
You are right @dotansimha ! I never used `withHOC` so I forgot we could do that 😬 Thanks for reminding me!

I have added that and also updated the tests to show explicitly which config (i.e. `withHOC`, `withHooks` or `withComponent`) we are trying to test in each

dotansimha commented:
Thank you @eddeee888 ! Looks great.
